### ISSUE TITLE
Obsoletion of terms with Reactome dbxrefs

### ISSUE THREAD

**Issue by pgaudet** [geneontology/go-ontology#17137]
Hello @cmungall @ukemi 
I am trying to obsolete a term with dbxrefs to Reactome; should I delete the dbxrefs ? These should be automatically be mapped to valid terms, right ? 

(I ran into this while doing #17071) 

Thanks, Pascale

**Comment by cmungall** [comment]
Looks like we didn't close the loop on this

reactome xrefs are imported, only axioms in go-edit.owl should be edited

But it looks like we redundantly retain xrefs in go-edit. Also the import file uses lowercase reactome

**Comment by pgaudet** [comment]
So I remove them from this term and I open a new ticket to deal with this globally ?

**Comment by deustp01** [comment]
At the Reactome end, curators reliably use currently valid GO molecular function terms (those are the only ones getting xrefs currently, I think) and before each release we run a script that updates our local copy of GO, flags GO terms that we have used that have become obsolete, makes an automatic update where there is a clean one-to-one replacement for the obsolete term, and generates an error message that goes to the curator who created the instance in all other cases. As a result, all Reactome-to-GO xrefs should link Reactome reaction instances to currently valid GO terms.

If that is not happening, there could be obsolete GO terms used in our annotations that are escaping our QA, in which case a list sent to me would be really helpful to get the bad instances and the QA gap fixed. Or it could be a problem with the script at the GO end that is supposed to drop all Reactome xrefs every time we do a new release and generate a complete new set based on the latest Reactome set of reaction instances and associated GO molecular function terms. An important corollary of this is that ALL Reactome xrefs should have identifiers of the form R-HSA-# where # is an integer string. Any xrefs with Reactome identifiers on the form REACT_# are obsolete and should have been dropped several months ago, so persistent identifiers of this form suggest a bug in the xref regeneration process at the GO end.

**Comment by ukemi** [comment]
It looks like this is dealt with on the Reactome side. The only issue is the rate at which we have been obsoleting terms makes it difficult to remain in synch. I think what we need to do here is filter out any xrefs from the load file that point to obsolete terms.

**Comment by deustp01** [comment]
Not sure that current GO obsoletion-fest is responsible for stale xrefs. I've been checking the lists of affected annotations; Reactome is rarely on the lists and when it is I've been able to make fixes.

**Comment by goodb** [comment]
I don't think Reactome is doing anything wrong here.  As I understand it, go-edit.obo should not ever have any reactome xrefs in it.  They should be added to the output files (go.obo go.owl etc.) by the make process.  The go-edit.obo file currently in Master has thousands of reactome xrefs in it.  Most of these use reactome:HSA##### ids but a small fraction use the more up to date Reactome:HSA### ids.  None use the old format of REACT: .   

It looks like one of the automated imports got merged into the edit file.  I think if we just go ahead and delete all the reactome xrefs in the current go-edit.obo file, the build will work properly.  

Any ideas how go.obo might have snuck its way into go-edit.obo ?  (Or am I missing something about how the edit/build process should work here?)

**Comment by goodb** [comment]
Sorry, my mistake. Opening the go-edit.obo file in Protege resulted in it loading the imported ontologies, including the reactome xrefs and this confused me.  In fact it looks like there are just 12 direct xrefs in the file that reference Reactome - all using the current pattern.  It looks like there are about 72 additional references within definitions, again all use the correct id pattern.  I believe when we migrated to the new pattern we decided to leave the xrefs within the definitions where they were and just replace the direct xrefs.  

Right now the import for the xrefs (and taxon constraints) is aiming at a file dated 2019-03-19.  It looks as though that build did not include the update to change from reactome the Reactome that was merged on the day before.  A fresh build should fix that particular problem.
